src/main/java/io/github/opencubicchunks/cubicchunks/core/asm/mixin/core/common/MixinWorld_HeightLimits.java

```diff
@@ -265,4 +268,16 @@ private boolean updateEntities_isChunkLoaded(World world, int chunkX, int chunkZ
             return this.isChunkLoaded(chunkX, chunkZ, allowEmpty);
         }
     }
+    
+    @Inject(method = "getBiome", at = @At("HEAD"), cancellable = true)
+    public void getBiome(BlockPos pos, CallbackInfoReturnable<Biome> ci) {
+        if (!this.isCubicWorld())
+            return;
+        ICube cube = this.getCubeCache().getLoadedCube(Coords.blockToCube(pos.getX()),Coords.blockToCube(pos.getY()),Coords.blockToCube(pos.getZ()));
+        if (cube == null)
+            return;
```

Comment:
if cube is null, try column

Reply:
Using 'return' here I will keep callback not cancelled, therefore "vanilla" function, which will get biome from chunk, will be called. Since cube is null there is no way to retrieve chunk faster, than using vanilla way.

Reply:
Then at least add a comment. If I got it wrong here, I (and probably someone else) is likely to get it wrong the next time.